app/WebParsing/ReqParser.hs

```diff
@@ -425,6 +430,30 @@ courseAsModParser = do
     req <- andParser courseParser
     return $ REQUIREMENT req
 
+-- | Parses a literal "course" or "courses"
+courseLiteralParser :: Parser String
+courseLiteralParser = Parsec.spaces
+    >> Parsec.choice (map caseInsensitiveStr [
+        "courses",
+        "course"
+        ])
+
+-- | Parses a department code in the fces requirement
+-- | eg. the "CSC" in "1.0 credit in CSC" or "1.0 credit in CSC courses"
+departmentParser :: Parser Modifier
+departmentParser = do
+    Parsec.spaces
+    Parsec.notFollowedBy fceSeparator
```

Comment:
Overall this and the next part are a bit too complicated. You can use `Parsec.manyTill`, and then explicitly fail if an empty string (or a string containing only whitespace) is consumed. You can use the `fail` function here this is technically part of the `MonadFail` type class, which `Parser` is a member of).